File: Data Structures & Algorithms/daily-temperatures/submission-1.cpp
class Solution {
public:

    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> backingStack;
        vector<int> result(temperatures.size());
        result[temperatures.size() - 1] = 0;
    

        for (int i = 0; i < temperatures.size(); i++) {
            if (i == 0) {
                //just add the first one
                backingStack.push(i);
            } else {
                //not the first one, regular logic. 
                while (!backingStack.empty() && temperatures[i] > temperatures[backingStack.top()]) {
                    int prevIndex = backingStack.top();
                    backingStack.pop();
                    result[prevIndex] = i - prevIndex;
                }
                //result[backingStack.top()] = count;
                backingStack.push(i);
            }

        }
        return result;
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        stack = []
        
        for bracket in s:
            if (bracket == '(') or (bracket == '{') or (bracket == '['):
                #open bracket
                stack.append(bracket)
            else:
                #closed bracket
                if (len(stack) == 0):
                    return False
                if (((bracket == ')') and (stack[-1] == '(')) or
                ((bracket == ']') and (stack[-1] == '[')) or
                ((bracket == '}') and (stack[-1] == '{'))):
                    stack.pop()
                else:
                    return False
        
        if (len(stack) == 0):
            return True
        else:
            return False
